openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -113,17 +132,36 @@ def dispatch(self, request, *args, **kwargs):
 
         return response
 
-    def _decompose_access_token_response(self, request, response):
+    def _decompose_access_token_response(self, adapter, response):
         """ Decomposes the access token in the request to an expiration date, scopes, and User. """
         content = json.loads(response.content)
         access_token = content['access_token']
         scope = content['scope']
-        access_token_obj = self.get_adapter(request).get_access_token(access_token)
-        user = access_token_obj.user
         scopes = scope.split(' ')
+        user = adapter.get_access_token(access_token).user
         expires_in = content['expires_in']
         return expires_in, scopes, user
 
+    def _get_client_specific_claims(self, client_id, adapter):
+        """ Get claims that are specific to the client. """
+        # If JWT scope enforcement is enabled, we need to sign tokens
+        # given to restricted application with a separate secret which
+        # other IDAs do not have access to. This prevents restricted
+        # applications from getting access to API endpoints available
+        # on other IDAs which have not yet been protected with the
+        # scope-related DRF permission classes. Once all endpoints have
+        # been protected we can remove this if/else and go back to using
+        # a single secret.
```

Comment:
Shall we add a TODO statement here with a link to a JIRA ticket?  So we remember to delete this code and comment when this happens?